### DISCUSSION TITLE
JWT response mode

### DISCUSSION THREAD

**Discussion by ziad696** [adrianbrs/nest-oidc-provider#21] [Q&A] [answered]
Can anyone show how to make access token in jwt format instead default one, which is opaque

**Comment by adrianbrs** [comment, marked as answer]
It's possible to customize the [`features.resourceIndicators.getResourceServerInfo`](https://github.com/panva/node-oidc-provider/blob/main/docs/README.md#getresourceserverinfo) configuration and change the access token format to `jwt`.

See the example below:
```ts
{
  features: {
    resourceIndicators: {
      getResourceServerInfo: () => {
        return {
          scope: 'api:read api:write',
          audience: 'resource-server-audience-value',
          accessTokenTTL: 2 * 60 * 60,
          accessTokenFormat: 'jwt', // JWT token format
          jwt: {
            sign: { alg: 'ES256' },
          },
        }
      }
    }
  }
}
```

BTW this config is from the `oidc-provider` package so you can open a [discussion in their repository](https://github.com/panva/node-oidc-provider/discussions) if you want.
